universe/Ship.cpp

```diff
@@ -272,11 +272,18 @@ bool Ship::HasFighters(const Universe& universe) const {
 bool Ship::CanColonize(const Universe& universe, const SpeciesManager& sm) const {
     if (m_species_name.empty())
         return false;
-    const Species* species = sm.GetSpecies(m_species_name);
-    if (!species || !species->CanColonize())
-        return false;
     const ShipDesign* design = universe.GetShipDesign(m_design_id);
-    return design && design->CanColonize(); // use design->CanColonize because zero-capacity colony ships still count as outpost ships, can "can colonize" as far as order / the UI are concerned
+    if (!design)
+        return false;
+    if (design && design->CanColonize()) {
```

Comment:
due to earlier "if (!design) return false" check, design here should always be true so no need to check second time. Or conversely drop the earlier check and count on the final "return false"